addons/account/models/account_payment.py

```diff
@@ -16,7 +16,7 @@ class AccountPayment(models.Model):
     move_id = fields.Many2one(
         comodel_name='account.move',
         string='Journal Entry', required=True, readonly=True, ondelete='cascade',
-        check_company=True)
+        check_company=True, index='btree')
```

Comment:
if you put it there, you probably need to put it as well on account.bank.statement.line